web/views.py

```
@@ -4644,71 +4690,464 @@
+        "is_teacher": is_teacher,
+        "is_enrolled": is_enrolled,
+        "enrolled_students": enrolled_students,
+        "present_students": present_students,
+        "teacher": classroom.teacher,
+    }
+
+    return render(request, "virtual_classroom/attendance.html", context)
+
+
+@login_required
+def update_student_attendance(request, classroom_id):
```

Ruff (0.11.9)

`update_student_attendance` is too complex (11 > 10)

(C901)